client/components/tinymce/plugins/contact-form/dialog/field.jsx

```diff
@@ -0,0 +1,100 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import PureRenderMixin from 'react-pure-render/mixin';
+import omit from 'lodash/object/omit';
+
+/**
+ * Internal dependencies
+ */
+import Gridicon from 'components/gridicon';
+import FieldHeader from './field-header';
+import FoldableCard from 'components/foldable-card';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormTextInput from 'components/forms/form-text-input';
+import FormCheckbox from 'components/forms/form-checkbox';
+import SelectDropdown from 'components/select-dropdown';
+import DropdownItem from 'components/select-dropdown/item';
+import TokenField from 'components/token-field';
+import getLabel from './locales';
+
+export default React.createClass( {
+	displayName: 'ContactFormDialogField',
+
+	mixins: [ PureRenderMixin ],
+
+	propTypes: {
+		label: PropTypes.string.isRequired,
+		type: PropTypes.string.isRequired,
+		options: PropTypes.string,
+		required: PropTypes.bool,
+		onUpdate: PropTypes.func.isRequired,
+		onRemove: PropTypes.func.isRequired
+	},
+
+	renderOptions() {
+		if ( this.props.type === 'radio' || this.props.type === 'select' ) {
+			let { options } = this.props;
+			if ( options && typeof options === 'string' ) {
+				options = options.split( ',' );
+			}
+
+			return (
+				<FormFieldset>
+					<FormLabel>{ this.translate( 'Options' ) }</FormLabel>
+					<TokenField
+						value={ options || [] }
+						onChange={ tokens => this.props.onUpdate( { options: tokens.join() } ) }/>
+				</FormFieldset>
+			);
+		}
+
+		return null;
```

Comment:
Two points on this function. First, it generally helps readability if you return early whenever possible, as it avoids the majority of the function being indented an extra level. Second, only the `render` method requires that a non-`undefined` value be returned, so `return null;` could be omitted altogether here (leveraging that functions without `return` implicitly `return undefined;`).

I think this function should be rewritten as:

```js
renderOptions() {
	if ( this.props.type !== 'radio' && this.props.type !== 'select' ) {
		return;
	}

	// ...

	return (
		<FormFieldSet>
			{ /* ... */ }
		</FormFieldSet>;
	);
}
```